File: day01/02.rb
f = File.open('input.txt', 'r')

arr = []

f.each_line do |line|
  arr << line.to_i
end
f.close

sum = 0
i = 0
mem = {}

loop do
  num = arr[i]
  if mem[sum]
    puts sum
    break
  else
    mem[sum] = true
    sum += num
    i = i + 1 == arr.size ? 0 : i + 1
  end
end

File: day01/02.py
file = open("input.txt", "r")

arr = []
map = {}

for line in file:
    arr.append(int(line))

i = 0
sum = 0

while True:
    sum += arr[i]
    if (sum in map):
        print(sum)
        break
    else:
        map[sum] = True

    if (i + 1) == len(arr):
        i = 0
    else:
        i += 1
